client/transport/src/main/java/org/elasticsearch/transport/client/PreBuiltTransportClient.java

```diff
@@ -79,13 +82,24 @@ public PreBuiltTransportClient(Settings settings, Collection<Class<? extends Plu
 
         @Override
         public Settings additionalSettings() {
-            return Settings.builder()
-                    .put(NetworkModule.TRANSPORT_TYPE_KEY, Netty4Plugin.NETTY_TRANSPORT_NAME)
-                    .put(NetworkModule.HTTP_TYPE_KEY, Netty4Plugin.NETTY_HTTP_TRANSPORT_NAME)
-                    .put("netty.assert.buglevel", true)
+            return Settings.builder().put("netty.assert.buglevel", true)
                     .build();
         }
 
     }
 
+    @Override
+    public void close() {
+        super.close();
+        if (NetworkModule.TRANSPORT_TYPE_SETTING.exists(settings) == false
+            || NetworkModule.TRANSPORT_TYPE_SETTING.get(settings).equals(Netty4Plugin.NETTY_TRANSPORT_NAME)) {
+            try {
+                GlobalEventExecutor.INSTANCE.awaitInactivity(5, TimeUnit.SECONDS);
+                ThreadDeathWatcher.awaitInactivity(5, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
```

Comment:
just restore the interrupt status here if you catch the exception and move on 

`Thread.currentThread().interrupt()` 

and don't throw RTExc